packages/aws-rfdk/lib/lambdas/nodejs/cert-rotation-monitor/handler.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* eslint-disable no-console */
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { SecretsManager, CloudWatch } from 'aws-sdk';
+
+import { Secret } from '../lib/secrets-manager';
+import { Certificate } from '../lib/x509-certs';
+
+const SECRETS_MANAGER_VERSION = '2017-10-17';
+
+export class CertificateRotationMonitor {
+  public static METRIC_NAME = 'DaysToExpiry';
+  public static METRIC_NAMESPACE = 'AWS/RFDK';
+  public static METRIC_DIMENSION = 'Certificate Metrics';
+
+  readonly uniqueID: string;
+
+  constructor(uniqueID: string) {
+    this.uniqueID = uniqueID;
+  }
+
+  private async getSecretsByTag(secretsManager: SecretsManager): Promise<SecretsManager.SecretListType | undefined> {
+    // Looking for all certificates that contain tags with uniqueID.
+    const request: SecretsManager.ListSecretsRequest = {
+      Filters: [{
+        Key: 'tag-value',
+        Values: [this.uniqueID],
+      }],
+    };
+    try {
+      const response: SecretsManager.ListSecretsResponse = await secretsManager.listSecrets(request).promise();
+      return response.SecretList;
+    } catch (e) {
+      throw new Error(`getSecretsByTag '${this.uniqueID}' failed':` +
+        `${e.code} -- ${e.message}`);
+    }
+  }
+
+  private putValueToMetric(value: number){
+    const metric = {
+      MetricData: [
+        {
+          MetricName: CertificateRotationMonitor.METRIC_NAME,
+          Dimensions: [
+            {
+              Name: CertificateRotationMonitor.METRIC_DIMENSION,
+              Value: this.uniqueID,
+            },
+          ],
+          Unit: 'None',
+          Value: value,
+        },
+      ],
+      Namespace: CertificateRotationMonitor.METRIC_NAMESPACE,
+    };
+    const cloudwatch = new CloudWatch();
+    cloudwatch.putMetricData(metric, (err) => {
+      if (err) {
+        console.log(`Failed to put value to CertDaysToExpiry metric: ${err.message}`);
+      } else {
+        console.log('Value was added to CertDaysToExpiry metric.');
+      }
+    });
+  }
```

Comment:
This should be changed into an `async` function, since `cloudwatch.putMetricData` is asynchronous and we want it to complete before the Lambda function exits.